dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java

```diff
@@ -283,6 +289,72 @@ private void parseInjectExtra(Element element, Map<TypeElement, InjectionTarget>
     erasedTargetTypes.add(erasedTargetType);
   }
 
+  private boolean isParcel(TypeMirror type) {
+    List<Element> supportedTypes = getTypeElements(
+            new Class[]{Byte.class, Double.class, Float.class, Integer.class,
+                        Long.class, Character.class, Boolean.class, String.class});
+
+    List<Element> singleGenericCollections = getTypeElements(
+            new Class[]{List.class, ArrayList.class, LinkedList.class,
+                        Set.class, HashSet.class, SortedSet.class, TreeSet.class,
+                        LinkedHashSet.class});
+
+    List<Element> doubleGenericCollections = getTypeElements(
+            new Class[]{Map.class, HashMap.class, LinkedHashMap.class,
+                        SortedMap.class, TreeMap.class});
+
+    return isParcelRecursive(
+            type,
+            supportedTypes,
+            singleGenericCollections,
+            doubleGenericCollections,
+            false);
+  }
+
+  private boolean isParcelRecursive(TypeMirror type,
+                                    List<Element> supportedTypes,
+                                    List<Element> singleGenericCollections,
+                                    List<Element> doubleGenericCollections,
+                                    boolean subCollection) {
+    if (subCollection && existsWithin(type, supportedTypes)) {
+      return true;
+    }
+    if (hasAnnotationWithFqcn(typeUtils.asElement(type), "org.parceler.Parcel")) {
```

Comment:
That does require that Parceler is on the classpath, the current technique allows for Parceler to be optional.